kernel-open/nvidia/nv-pci.c

```diff
@@ -223,7 +223,79 @@ static void nv_init_dynamic_power_management
 
 
 
+static int nv_resize_pcie_bars(struct pci_dev *pci_dev) {
+    struct pci_host_bridge *host;
+    u16 cmd;
+    int r, old_size, requested_size;
+    int ret = 0;
+
+    // Check if BAR1 has PCIe rebar capabilities
+    u32 sizes = pci_rebar_get_possible_sizes(pci_dev, NV_GPU_BAR1);
```

Comment:
pci_rebar_get_possible_sizes() was added in v5.12 by commit 8fbdbb66f8c10. In order to avoid breaking the build on earlier kernels, a test for this function should be added to conftest.sh.

The kernel function has to be called without any arguments for the test to work, which is counterintuitive at first. The test should look something like this:
```
pci_rebar_get_possible_sizes)
            #
            # Determine if the pci_rebar_get_possible_sizes() function is present.
            #
            # Added by commit 8fbdbb66f8c10 ("PCI: Add resizable BAR infrastructure
            # ") in v5.12
            #
            CODE="
            #include <linux/pci.h>
            void conftest_pci_rebar_get_possible_sizes(void) {
                pci_rebar_get_possible_sizes();
            }"

            compile_check_conftest "$CODE" "NV_PCI_REBAR_GET_POSSIBLE_SIZES_PRESENT" "" "functions"
        ;;
```
It should also be added to nvidia.Kbuild:
```
NV_CONFTEST_FUNCTION_COMPILE_TESTS += pci_rebar_get_possible_sizes
```
After this, #if defined(NV_PCI_REBAR_GET_POSSIBLE_SIZES_PRESENT) can be used to either build this function, or make it a no-op for unsupported kernels (with possibly a helpful info or debug message).